lessons/cosmosdb.md

```diff
@@ -32,8 +43,339 @@ In this exercise we'll look into Cosmos DB Emulator to see how you can interact
+
+```
+
+After adding the output binding the signature of the Run method should look like the following:
+
+```csharp
+ public static void Run([QueueTrigger("newplayer-items", Connection = "queueConnection")]string myQueueItem,
+                [CosmosDB(
+                databaseName: "Players",
+                collectionName: "players",
+                ConnectionStringSetting = "CosmosDBConnection")]out dynamic document,
+                 ILogger log)
+```
+
+Save your changes.
+
+Add a Player.cs model to the Azure Function application.
+
+For this exercise, we need to map the item from the queue to an Entity that can be deserialized to a C# object, and make the required transformations, and then deserialize this object to save it in CosmosDB.
+
+Add a new folder called Models, and add a new C# class inside of it. Call it Player.
+
+Add the following content to this class.
+
+```csharp
+
+using Newtonsoft.Json;
+
+namespace AzureFunctionsUniversity.Demo.Cosmos.Models
+{
+    public class Player
+    {
+        public Player()
+        {
+        }
+
+        [JsonProperty("id")]
+        public string Id { get; set; }
+        
+        [JsonProperty("nickName")]
+        public string NickName { get; set; }
+        
+        [JsonProperty("playerId")]
+        public int playerId { get; set; }
+        
+        [JsonProperty("region")]
+        public string Region { get; set; }
+    }
+}
+
+```
+
+Save your changes.
+
+2.9 Modify the content of the Run method.
+
+At the Run method, get the message from the queue, do a simple transformation and then send the resulting data to Cosmos DB.
+
+The final code is shown below:
+
+```csharp
+public static class QueueTriggerCSharp1
+    {
+        [FunctionName("QueueTriggerCSharp1")]
+        public static void Run([QueueTrigger("newplayer-items", Connection = "queueConnection")]string myQueueItem,
+                [CosmosDB(
+                databaseName: "Players",
+                collectionName: "players",
+                ConnectionStringSetting = "CosmosDBConnection")]out dynamic document,
+                 ILogger log)
+        {            
+            log.LogInformation($"C# Queue trigger function processed: {myQueueItem}");
+            Player player = JsonConvert.DeserializeObject<Player>(myQueueItem);
+
+            //The code for the data transformation should be added here
+            player.NickName = player.NickName.ToUpperInvariant();
+            
+            // return the player data in the document variable used by the output binding 
+            document = player;
+
+            log.LogInformation($"C# Queue trigger function inserted one row");
+            log.LogInformation($"Description={myQueueItem}");
+
+        }
+    }
+
+```
+
+Make sure to add a reference to the `Microsoft.Azure.WebJobs.Extensions.CosmosDB` package. For adding the package to your app, run the `dotnet add package Microsoft.Azure.WebJobs.Extensions.CosmosDB` command.

```

Comment:
Perhaps move the addition of the Cosmos package a bit earlier before we add the Cosmos binding code?

Reply:
Sure!